revenue_app/static/revenue_app/js/generate_pdf.js

```diff
@@ -0,0 +1,66 @@
+function generatePDF() {
+
+  //Get svg markup as string
+  var titleArs = document.getElementById('title-ars');
+  var tableArs = document.getElementById('dynamic-table-ars');
+  var svgChartArs = document.getElementById('chart-top-ars').innerHTML;
+  var svgLegendArs = document.getElementById('legend-top-ars').innerHTML;
+
+  svgChartArs = svgChartArs.replace(/\r?\n|\r/g, '').trim();
+  svglegend = svgLegendArs.replace(/\r?\n|\r/g, '').trim();
+
+  var canvasChartArs = document.createElement('canvas');
+  var canvasLegendArs = document.createElement('canvas');
+  var context1 = canvasChartArs.getContext('2d');
+  var context2 = canvasLegendArs.getContext('2d');
+
+  context1.clearRect(0, 0, canvasChartArs.width, canvasChartArs.height);
+  context2.clearRect(0, 0, canvasLegendArs.width, canvasLegendArs.height);
+  canvg(canvasChartArs, svgChartArs);
+  canvg(canvasLegendArs, svgLegendArs);
+
+  var imgChartArs = canvasChartArs.toDataURL('image/png');
+  var imgLegendArs = canvasLegendArs.toDataURL('image/png');
+
+  var titleBrl = document.getElementById('title-brl');
+  var tableBrl = document.getElementById('dynamic-table-brl');
+  var svgChartBrl = document.getElementById('chart-top-brl').innerHTML;
+  var svgLegendBrl = document.getElementById('legend-top-brl').innerHTML;
+
+  svgChartBrl = svgChartBrl.replace(/\r?\n|\r/g, '').trim();
+  svglegend = svgLegendBrl.replace(/\r?\n|\r/g, '').trim();
+
+  var canvasChartBrl = document.createElement('canvas');
+  var canvasLegendBrl = document.createElement('canvas');
+  var context1 = canvasChartBrl.getContext('2d');
+  var context2 = canvasLegendBrl.getContext('2d');
+
+  context1.clearRect(0, 0, canvasChartBrl.width, canvasChartBrl.height);
+  context2.clearRect(0, 0, canvasLegendBrl.width, canvasLegendBrl.height);
+  canvg(canvasChartBrl, svgChartBrl);
+  canvg(canvasLegendBrl, svgLegendBrl);
+
+  var imgChartBrl = canvasChartBrl.toDataURL('image/png');
+  var imgLegendBrl = canvasLegendBrl.toDataURL('image/png');
```

Comment:
This is repeated code, may be we can think how to do this with a loop or a function